src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -6155,16 +6155,17 @@ class NotPointer
             CreateStandardCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
                 // (9,26): error CS0254: The right hand side of a fixed statement assignment may not be a cast expression
                 //         fixed (byte* p = (byte*)&x)
-                Diagnostic(ErrorCode.ERR_BadCastInFixed, "(byte*)&x"),
+                Diagnostic(ErrorCode.ERR_BadCastInFixed, "(byte*)&x").WithLocation(9, 26),
                 // (13,25): error CS0213: You cannot use the fixed statement to take the address of an already fixed expression
                 //         fixed (int* p = n) //CS0213 (confusing, but matches dev10)
-                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "n"),
+                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "n").WithLocation(13, 25),
+                // (13,25): error CS0266: Cannot implicitly convert type 'NotPointer*' to 'int*'. An explicit conversion exists (are you missing a cast?)
+                //         fixed (int* p = n) //CS0213 (confusing, but matches dev10)
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "n").WithArguments("NotPointer*", "int*").WithLocation(13, 25),
```

Comment:
It is hard to say if  `NotPointer* -> int*` conversion exists since we would normally not allow  such types as `NotPointer*`.

I will check if this error is easy to suppress. The quality expectations for error messages in broken unsafe code is evidently not very high though.

---
In reply to: [124914123](https://github.com/dotnet/roslyn/pull/20548#discussion_r124914123) [](ancestors = 124914123)